functests/ptp/ptp.go

```diff
@@ -108,4 +93,417 @@ var _ = Describe("ptp", func() {
+				profileMaster := "Profile Name: grandmaster"
+				for _, pod := range ptpRunningPods {
+					podLogs, err := getPodLogs(pod)
+					Expect(err).NotTo(HaveOccurred(), "Error to find needed log due to %s", err)
+					if podRole(pod, ptpSlaveNodeLabel) {
+						Expect(podLogs).Should(ContainSubstring(profileSlave),
+							fmt.Sprintf("Profile \"%s\" not found in pod's log %s", profileSlave, pod.Name))
+					}
+					if podRole(pod, ptpGrandmasterNodeLabel) {
+						Expect(podLogs).Should(ContainSubstring(profileMaster),
+							fmt.Sprintf("Profile \"%s\" not found in pod's log %s", profileSlave, pod.Name))
+					}
+				}
+			})
+
+			// 25738
+			It("Slave can sync to master", func() {
+				var masterID string
+				var slaveMasterID string
+				grandMaster := "assuming the grand master role"
+				for _, pod := range ptpRunningPods {
+					if podRole(pod, ptpGrandmasterNodeLabel) {
+						podLogs, err := getPodLogs(pod)
+						Expect(err).NotTo(HaveOccurred(), "Error to find needed log due to %s", err)
+						Expect(podLogs).Should(ContainSubstring(grandMaster),
+							fmt.Sprintf("Log message \"%s\" not found in pod's log %s", grandMaster, pod.Name))
+						for _, line := range strings.Split(podLogs, "\n") {
+							if strings.Contains(line, "selected local clock") && strings.Contains(line, "as best master") {
+								masterID = strings.Split(line, " ")[4]
+							}
+						}
+					}
+					if podRole(pod, ptpSlaveNodeLabel) {
+						podLogs, err := getPodLogs(pod)
+						Expect(err).NotTo(HaveOccurred(), "Error to find needed log due to %s", err)
+						for _, line := range strings.Split(podLogs, "\n") {
+							if strings.Contains(line, "new foreign master") {
+								slaveMasterID = strings.Split(line, " ")[6]
+							}
+						}
+					}
+				}
+				Expect(masterID).NotTo(BeNil())
+				Expect(slaveMasterID).NotTo(BeNil())
+				Expect(slaveMasterID).Should(HavePrefix(masterID), "Error match MasterID with the SlaveID. Slave connected to another Master")
+			})
+		})
+	})
+	var _ = Describe("Test Offset", func() {
+		BeforeEach(func() {
+			nodes, err := client.Client.Nodes().List(metav1.ListOptions{
+				LabelSelector: ptpSlaveNodeLabel,
+			})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(nodes.Items)).To(BeNumerically(">", 0))
+		})
+
+		var _ = Context("PTP configuration verifications", func() {
+
+			// 27324
+			It("PTP time diff between Grandmaster and Slave should be in range -100ms and 100ms", func() {
+				var timeDiff string
+				ptpPods, err := client.Client.Pods(ptpLinuxDaemonNamespace).List(metav1.ListOptions{LabelSelector: "app=linuxptp-daemon"})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(len(ptpPods.Items)).To(BeNumerically(">", 0))
+				slavePodDetected := false
+				for _, pod := range ptpPods.Items {
+					if podRole(pod, ptpSlaveNodeLabel) {
+						Eventually(func() string {
+							buf, _ := pods.ExecCommand(client.Client, pod, []string{"curl", "127.0.0.1:9091/metrics"})
```

Comment:
Same as above, why do we need to wait?

Reply:
The issues is test cycles. If you finish the first cycle and want to run new one then you have to wait until ptp pods syncs with each other.

Reply:
Not sure I understand this. The tests are supposed to run when everything is settled and the deploy is complete.
We are not running new pods in tests here if I understood it correctly.

Reply:
We decided to leave it as it is